File: VB/VB/WindowsApplication1/RibbonForm1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraBars
Imports DevExpress.XtraEditors.Repository
Imports System.Threading
Imports DevExpress.XtraEditors

Namespace WindowsApplication1
	Partial Public Class RibbonForm1
		Inherits DevExpress.XtraBars.Ribbon.RibbonForm
		Public Sub New()
			InitializeComponent()
			InitValues()
			Calculate()
			For Each item As RepositoryItem In ribbon.Manager.RepositoryItems
				AddHandler item.EditValueChanged, AddressOf item_EditValueChanged
			Next item
		End Sub

		Private Sub item_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
			If TypeOf sender Is CheckEdit Then
				If (TryCast(sender, CheckEdit)).Checked Then
					beiProgress.Visibility = BarItemVisibility.Always
				Else
					beiProgress.Visibility = BarItemVisibility.Never
				End If
				Return
			End If
			ribbon.Manager.ActiveEditItemLink.PostEditor()
			ShowProgress()
			Calculate()
		End Sub
		Private Sub InitValues()
			beiValueA.EditValue = 1
			beiOperation.EditValue = "Plus"
			beiValueB.EditValue = 2
			beiShowProgress.EditValue = True
		End Sub

		Private Sub ShowProgress()
			If beiProgress.Visibility = BarItemVisibility.Never Then
				Return
			End If
			For i As Integer = repositoryItemProgressBar1.Minimum To repositoryItemProgressBar1.Maximum - 1
				beiProgress.EditValue = i
				Application.DoEvents()
				Thread.Sleep(1)
			Next i
		End Sub
		Private Sub Calculate()
			Dim operation As Integer
			If Object.Equals(beiOperation.EditValue, "Minus") Then
				operation = -1
			Else
				operation = 1
			End If
			Dim result As Single = Convert.ToInt32(beiValueA.EditValue) + Convert.ToInt32(beiValueB.EditValue) * operation
			bsiResult.Caption ="Result = " & result
		End Sub
	End Class
End Namespace

File: CS/WindowsApplication1/RibbonForm1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraBars;
using DevExpress.XtraEditors.Repository;
using System.Threading;
using DevExpress.XtraEditors;

namespace WindowsApplication1
{
    public partial class RibbonForm1 : DevExpress.XtraBars.Ribbon.RibbonForm
    {
        public RibbonForm1()
        {
            InitializeComponent();
            InitValues();
            Calculate();
            foreach (RepositoryItem item in ribbon.Manager.RepositoryItems)
            {
                item.EditValueChanged += new EventHandler(item_EditValueChanged);
            }
        }

        void item_EditValueChanged(object sender, EventArgs e)
        {
            if (sender is CheckEdit)
            {
                beiProgress.Visibility = (sender as CheckEdit).Checked ? BarItemVisibility.Always : BarItemVisibility.Never;
                return;
            }
            ribbon.Manager.ActiveEditItemLink.PostEditor();
            ShowProgress();
            Calculate();
        }
        private void InitValues()
        {
            beiValueA.EditValue = 1;
            beiOperation.EditValue = "Plus";
            beiValueB.EditValue = 2;
            beiShowProgress.EditValue = true;           
        }

        private void ShowProgress()
        {
            if (beiProgress.Visibility == BarItemVisibility.Never)
                return;
            for (int i = repositoryItemProgressBar1.Minimum; i < repositoryItemProgressBar1.Maximum; i++)
            {
                beiProgress.EditValue = i;
                Application.DoEvents();
                Thread.Sleep(1);
            }
        }
        private void Calculate()
        {
            int operation = object.Equals(beiOperation.EditValue, "Minus") ? -1 : 1;
            float result = Convert.ToInt32(beiValueA.EditValue) + Convert.ToInt32(beiValueB.EditValue) * operation;
            bsiResult.Caption ="Result = " +  result;
        }
    }
}